### DISCUSSION TITLE
How to work with Jupiter Notebooks.

### DISCUSSION THREAD

**Discussion by nepomnyi** [alexlib/pivpy#51] [Q&A] [answered]
Hello, dear PIVPY community.
I am making a tutorial for VortexFitting package. I am having a hard time with Jupiter Notebooks. I don't know how to work with them. 
The only thing I know about them is that they exist in the form of websites and that I have to go to the Internet to work with them.
But PIVPY uses them as some files stored in the Examples folder. That confuses me a lot. 
OpenPIV documentation has [this piece](https://openpiv.readthedocs.io/en/latest/src/developers.html) describing very briefly how to work with them. But PIVPY documentation doesn't have it. When I use OpenPIV's commands with PIVPY package, I am running into an error. 
I am wondering if somebody can be so kind, to create more detailed instructions for PIVPY package on how to make these Jupiter notebooks, please.
Thank you in advance.
Ivan

**Comment by alexlib** [comment, marked as answer]
# what is it about:
Jupyter notebooks are only a tool, one does not have to use them to run `pivpy` or to use it. It's more convenient in my view for research purposes and interactive analysis and not for batch of large number of files. Today modern IDE with Python code allow to use small chunks of code (cells) sequentially and not running the full complete script at once, plus they allow to visualize the results quickly. In this sense, both IDE and Jupyter notebook give the same result. Jupyter notebooks have several additional benefits, e.g. using ipywidgets and mouse interactions, but these are aside now. 

# How to use Jupyter notebooks without installation: 
1. Use Google colab: https://colab.research.google.com/ 
2. Use Jupyterlite or Binder: https://jupyter.org/try  or https://jupyter.org/try-jupyter/lab/ 


# How to install Jupyter locally: 
1. It's recommended to use virtual environments (not to mix up different Python packages and versions) - e.g. `conda` or `mamba` or others, e.g. `venv`
2. Install it using `pip install jupyter` or `conda install jupyter`
3. Run: `jupyter notebook <name of your notebook>.ipynb`

**Reply by nepomnyi** [reply]
Thanks a million!
I was more concerned with how to run Jupiter locally, on a computer. The instructions I found in the OpenPIV documentation were driving me crazy. But these ones are so concise and look easy to work with. I'll try them on the weekend and see how it goes, thank you.

**Reply by nepomnyi** [reply]
I ended up using `conda install anaconda::jupyter` from [Anaconda website](https://anaconda.org/anaconda/jupyter) because neither `pip install jupyter`, nor `conda install jupyter` worked.
With regards to running the notebook from console,` jupyter notebook <name of your notebook>.ipynb` didn't work either and I couldn't figure out the working way. Instead, my VS Code offers a way to run Jupiter notebooks, that's what I ended up using.
